src/cloudevent/mocks/database/database-on-ref-created.ts

```diff
@@ -0,0 +1,57 @@
+import { DeepPartial, MockCloudEventAbstractFactory } from '../../types';
+import { CloudEvent, CloudFunction, database } from 'firebase-functions/v2';
+import { getBaseCloudEvent, getEventType, makeDataSnapshot } from '../helpers';
+
+export const databaseOnRefCreated: MockCloudEventAbstractFactory<database.DatabaseEvent<
+  database.DataSnapshot
+>> = {
+  generateMock(
+    cloudFunction: CloudFunction<database.DatabaseEvent<any>>,
+    cloudEventPartial?: DeepPartial<database.DatabaseEvent<any>>
+  ): database.DatabaseEvent<any> {
+    const instance = (cloudEventPartial?.instance as string) || 'instance-1';
```

Comment:
What about the stuff that comes from the cloud function? E.g. their instance? their ref?

Also, when it comes to the ref from the function, you'll need to inject values for the captured params. 

E.g. function.__endpoint.eventTrigger.eventFilters[ref] == "/users/{uid}"

would have an event ref of "/users/12345" and params would have {"uid": "12345"}

Reply:
Thanks for calling that out!!! I've updated each of these to check the function for the ref and instance.

Generating the params from the raw ref strings is a little bit tricker. I'm still working through either clone-n-own or rolling a custom solution..